DMCompiler/DM/Visitors/DMVisitorExpression.cs

```diff
@@ -183,12 +184,48 @@ public void VisitIdentifier(DMASTIdentifier identifier)
             // arglist hack
             if (procCall.Callable is DMASTCallableProcIdentifier ident) {
                 if (ident.Identifier == "arglist") {
-                    if (procCall.Parameters.Length != 1) throw new CompileErrorException(procCall.Location, "arglist must have 1 argument");
+                    switch (procCall.Parameters.Length) {
+                        case 0:
+                            DMCompiler.Emit(WarningCode.BadArgument, procCall.Location, "arglist() requires 1 argument");
+                            break;
+                        case 1:
+                            break;
+                        default:
+                            DMCompiler.Emit(
+                                WarningCode.TooManyArguments,
+                                procCall.Location,
+                                $"arglist() given {procCall.Parameters.Length} arguments, expecting 1");
+                            break;
+                    }
 
                     var expr = DMExpression.Create(_dmObject, _proc, procCall.Parameters[0].Value, _inferredPath);
                     Result = new Expressions.Arglist(procCall.Location, expr);
                     return;
                 }
+                if (!DMCompiler.Settings.NoStandard && ident.Identifier == "matrix") {
+                    // This is a simple lint against weird /matrix() calls
+                    // BYOND does compiletime on some of these, too, so might as well do our own linting :^)
+                    switch (procCall.Parameters.Length) {
+                        case 6: // Normal cases
+                        case 1:
+                        case 0:
+                            break;
+                        case 2: // These imply that they're trying to use the undocumented matrix signatures.
+                        case 3: // The lint is to just check that the last argument is a numeric constant that is a valid matrix "opcode."
+                        case 4:
+                            var lastArgument = procCall.Parameters.Last().Value;
+                            if(lastArgument is not DMASTConstantInteger) {
```

Comment:
This would cause problems if a non-constant was being used. If you want to put the effort into it, maybe instead make a special matrix DMExpression and handle the emissions in there where more information is available?

Reply:
That is true. My idea for this was to more provide an "are you sure?" sort of warning, rather than something that definitely proves a bad call.

In general, I can't really imagine a situation where a target would save the opcode literal into a variable and then access it to invoke the opcode signatures. I ran my branch against my copy of Paradise and this warning was never emitted.

I'd be okay with downgrading this into just a notice in the default config, though, if you're worried about emitting false positives.

Reply:
After talking, I've decided to move this behaviour over into the `ProcCall` DMExpression. Later on, we can place more compiletime linting of builtins in there :^)